tools/perf/builtin-c2c.c

```diff
@@ -2547,9 +3139,814 @@ static void perf_c2c__hists_fprintf(FILE *out, struct perf_session *session)
 	fprintf(out, "#\n");
 
 	print_pareto(out, perf_session__env(session));
+
+	/* Add symbol view in stdio mode */
+	if (build_symbol_hists(perf_session__env(session)) == 0) {
+		/* Count the number of entries */
+		int symbol_entries = 0;
+		struct rb_node *nd = rb_first_cached(&c2c.symbol_hists.hists.entries);
+		while (nd) {
+			struct hist_entry *he = rb_entry(nd, struct hist_entry, rb_node);
+			if (!he->filtered)
+				symbol_entries++;
+			nd = rb_next(nd);
+		}
+
+		fprintf(out, "\n");
+		fprintf(out, "Shared Data Symbols Table     (%d entries, sorted on Cycles Percent)\n", symbol_entries);
+		fprintf(out, "#\n");
+
+		hists__fprintf(&c2c.symbol_hists.hists, true, 0, 0, 0, stdout, true);
+	}
 }
 
 #ifdef HAVE_SLANG_SUPPORT
+
+/*
+ * Create child hist_entry objects for related symbols
+ * This enables the hierarchical display in the TUI browser
+ * Children are inserted sorted by Stores (stats.store) descending
+ */
+static void populate_symbol_children(struct hist_entry *he)
```

Comment:
the function is too long, should split into smaller function to improve readability.